### ISSUE TITLE
Hybrid search duplicates documents

### ISSUE THREAD

**Issue by hansand02** [meilisearch/meilisearch#5045]
Meilisearch v: 1.10.3
Hey! We are currently implementing a new search experience with Meilisearch.
We encountered an issue where the pagination of Meilisearch breaks. For every next page we fetch the result will include all the prior hits, including any new hits.
In short:
- The first element on each page from page 1-n, is the same element. ( see object below ). When setting the value of page, the first element in the result is the same no matter which page I choose.
- This behaviour only happens the semantic ratio is set to a number between 0.1-0.9. If i set it to 1, the embedder throws the error that OPENAI_API_KEY is missing.
**How to reproduce:**
In an index with vector search enabled:
- Insert an invalid/outdated api key to create an invalid embedder
- Set a suffiently low/high semantic ratio (0.5 worked to produce the issue for us )
- Choose first page through page: number, in meiliSearchParams
- Inspect that the first item is the same in each successive page.
Object with valid api key (masked)
``` json
{
    “embedder”: {
        “source”: “openAi”,
        “model”: “text-embedding-3-small”,
        “apiKey”: “sk-prXXXXXX...“,
        “dimensions”: 512,
        “documentTemplate”: “{{ doc._embeddingString }}”
    }
}
```

**Comment by qasimgulzar** [comment]
If someone has not already started working on this issue, I would be happy to take care of it.

**Comment by dureuill** [comment]
Hello @hansand02 

Thank you for your report, I opened a PR (#5063) with a fix.

@qasimgulzar : thank you for proposing your help. Here the fix was extra small, so I simply opened the PR

**Comment by curquiza** [comment]
Fixed by #5063
